### ISSUE TITLE
P0 - Documentation ready for Beta Release

### ISSUE THREAD

**Issue by scarlett2018** [microsoft/pai#4023]
## Checklist for public release

We plan to separate the docs into 4 parts:

  - **Manual for data scientists**: This manual is for data scientists to learn how to submit job, debug job, manage data, use Marketplace, VSCode / Jupyter extension and CLI support on OpenPAI.
  - **Manual for cluster administrators**: This manual is for cluster administrators to learn the installation and uninstallation of OpenPAI, some basic management operations, storage management, troubleshootiong, etc.
  - **Manual for PAI developers**: This manual is for potential PAI developers to quickly go through the achitecture of OpenPAI and learn how to leverage the SDK and develop new plugins.
  - **Design documentation (Reference)**: PAI design documentation and API reference for different PAI components. 

I have created a branch `user_manual` for creating these 3 manuals, the link to the manual is https://github.com/microsoft/pai/tree/user_manual/manual . The planned contents of these manuals are as follows:

### Contents for data scientists manual
- [x] Quick Start
  - [x] [Submit a hello world job](https://github.com/microsoft/pai/blob/master/docs/user/job_submission.md)
  - [x] Browse stdout, stderr, full logs, metrics
- [x] Work with docker image
   - [x] [Introduction to pre-built docker images](https://github.com/microsoft/pai/tree/zhiyuhe/edu/standard_dockers/contrib/standard-dockers)
- [x] How to Manage Data (***need help***) @Binyang2014 
  - [x] Team-wise storage - NFS
  - [x] Team-wise storage - AzureFile
  - [x] Team-wise storage - AzureBlob (create an issue for the current known problem)
  - Current doc: https://github.com/microsoft/pai/blob/master/docs/setup-persistent-volumes-on-pai.md
- [x] How to Debug Job
  - [x] [How to use SSH plugin](https://github.com/microsoft/pai/tree/master/src/kube-runtime/src/plugins/ssh)
  - [x] [How to use TensorBoard plugin](https://github.com/microsoft/pai/blob/master/docs/user/tensorboard.md)
- [x] Advanced Jobs
  - [x] Retry policy and [job exit spec](https://github.com/microsoft/pai/blob/master/src/k8s-job-exit-spec/config/user-manual.md) (***need help***) @yqwang-ms 
  - [x] Job example with multiple task roles 
     - [x] [tensorflow_cifar10](https://github.com/microsoft/pai/blob/master/marketplace-v2/tensorflow-cifar10.yaml)
     - [x] [horovod_pytorch](https://github.com/microsoft/pai/blob/master/marketplace-v2/horovod-pytorch-synthetic-benchmark.yaml)
  - [x] [Port reservation , environmental variables, parameters, and secrets](https://github.com/microsoft/pai/blob/master/docs/user/job_submission.md#use-parameters-and-secrets)
  - [x] InfiniBand job (***need help***) @abuccts 
  - [x] [Reference link to job protocol](https://github.com/microsoft/openpai-protocol/blob/master/schemas/v2/schema.yaml)
- [x] Use Marketplace
  - [x] Brief usage introduction (***need help***) @debuggy 
- [x] Use VSCode extension
  - [x] [Brief usage introduction](https://github.com/microsoft/openpaivscode)
  - [x] [Reference link to VSCode extension](https://github.com/microsoft/openpaivscode)
- [x] Use Jupyter notebook extension
  - [x] [Jupyter notebook extension](https://github.com/microsoft/openpai-nb)

### Contents for cluster administrator manual

- [x] Installation Guide (Quick Start)
  - [x] [From scratch](https://github.com/microsoft/pai/blob/master/contrib/kubespray/doc/quick-start.md)
- [x] Major breaking changes comparing with previous version 
  - [x] [From previous deployment](https://github.com/microsoft/pai/blob/master/docs/upgrade/upgrade_to_v0.18.0.md)
- [x] Installation FAQ (***need help***)  @abuccts
  - [x] Questions about Installation
  - [x] Installation troubleshooting 
- [x] Basic Management Operations
  - [x] Mangement on Webportal @hzy46 
    - [x] Hardware utilization page
    - [x] Services page
    - [x] User management
    - [x] Abnormal jobs
    - [x] [How to access k8s dashboard](https://github.com/microsoft/pai/blob/master/docs/k8s-dashboard/access-k8s-dashboard.md)
  - [x] PAI service management and paictl
    - [x] Brief introduction to PAI services
    - [x] [Use paictl to customize config, and restart PAI services](https://github.com/microsoft/pai/blob/master/docs/paictl/paictl-manual.md)
- [x] How to Manage Users and Groups
  - [x] [users and groups in basic authentication mode](https://github.com/wutb15/pai/blob/b80c8dffd8e028e0013366eb0cc7d622ae0c9917/docs/user_group_management.md) (***need help***) @debuggy 
    - [x] how to access rest-server
    - [x] how to use rest-server to configure user and groups in baisc auth mode
  - [x] users and groups in AAD mode
    - [x] [How to Setup AAD](https://github.com/microsoft/pai/blob/master/docs/aad-e2e/aad-e2e.md)
- [x] How to Setup Team-wise Storage  (***need help***) @Binyang2014 
  - [x] NFS server (storage manager)
  - [x] Samba 
  - [x] AzureFile
  - [x] Azureblob
- [x] How to Setup Virtual Clusters
  - [x] Introduction to hivedscheduler and how to configure it
  - [x] Example: single-type GPUs in different VCs
  - [x] Example: multi-type GPUs in different VCs
  - [x] How to grant VC to user groups (both AAD mode and auth mode)
- [x] How to Add / Remove Node (***need help***) @ydye 
   - [x] Guide to add / remove nodes (add link to "How to Setup Virtual Clusters" because hivedscheduler config needs change, too)
   - [x] Example: setup multi-type GPUs and split them into multiple Virtual Clusters
- [x] How to use CPU Nodes
  - [x] How to add CPU nodes and setup CPU-only VC
  - [x] How to submit CPU-only job
- [x] How to customize cluster by plugins
  - [x] [how to enable (install) a webportal plugin](https://github.com/microsoft/pai/blob/d659ffd97dedfc23ce1b06eebc2911b9c6592ac7/docs/webportal/PLUGINS.md)
- [x] Troubleshooting (***need help***)
  - [x] What if GPU is not detected
  - [x] What if some node is lost  
  - [x] What if some PAI service is not working
- [x] How to Uninstall OpenPAI
- [x] Upgrade Guide (needed in future)

### Contents for PAI developer Manual

- [ ] [OpenPAI Basic Design and Architecture](https://github.com/microsoft/pai/blob/141301d4ad637cbe0fff41e40e60bf7bf2797900/docs/system_architecture.md)
  - [ ] OpenPAI and Kubernetes (e.g. Framework Controller)
    - [ ] How OpenPAI uses Kubernetes 
    - [ ] Lifetime Management of OpenPAI Jobs (***need help***) @yifan
  - [ ] OpenPAI optimization for deep learning jobs (***need help***)
    - [ ] How OpenPAI uses Nvidia-docker
- [ ] Developer interface
  - [ ] [Docs] OpenAPI docs for RESTful API #3996
  - [ ] Use SDK
    - [ ] [Brief Usage introduction](https://github.com/microsoft/openpaisdk)
    - [ ] [Reference link to SDK](https://github.com/microsoft/openpaisdk)
- [ ] How OpenPAI Manages Storage
- [ ] Network Management of Mixed Cloud Environment (***need help***) @fanyangCS 
- [ ] How to Develop a Runtime Plugin (***need help***) @yqwang-ms 
- [ ] [How to Develop a Webportal Plugin] @Binyang2014 (https://github.com/microsoft/pai/blob/d659ffd97dedfc23ce1b06eebc2911b9c6592ac7/docs/webportal/PLUGINS.md#provide-plugin-configurations)
- [ ] How to Configure Framework Controller (***need help***) @yqwang-ms 
- [ ] How to Customize Job Protocol  (***need help***) @abuccts 
- [ ] How to Develop a Scheduler (***need help***) @yqwang-ms 

### Contents for design documentation (to be updated @hzy46 )
- [ ] OpenPAI overview
- [ ] OpenPAI architecture and component introduction

**Comment by fanyangCS** [comment]
related https://github.com/microsoft/pai/issues/2130

**Comment by fanyangCS** [comment]
https://github.com/microsoft/pai/issues/1763

**Comment by hzy46** [comment]
### Design

We plan to separate the docs into 4 parts:

  - **Manual for cluster users**: This manual is for cluster users to learn how to submit job, debug job, manage data, use Marketplace, VSCode extension and SDK support on OpenPAI.
  - **Manual for cluster administrators**: This manual is for cluster administrators to learn the installation and uninstallation of OpenPAI, some basic management operations, storage management, troubleshootiong, etc.
  - **Manual for PAI developers**: This manual is for potential PAI developers to quickly go through the achitecture of OpenPAI and learn how to develop new plugins.
  - **Design documentation (Reference)**: PAI design documentation and API reference for different PAI components. 

### Content of cluster user manual
- [ ] Quick Start
  - [ ] [Submit a hello world job](https://github.com/microsoft/pai/blob/master/docs/user/job_submission.md)
  - [ ] See stdout, stderr, full logs, metrics
  - [ ] [Introduction to pre-built docker images](https://github.com/microsoft/pai/tree/zhiyuhe/edu/standard_dockers/contrib/standard-dockers)
- [ ] How to Debug Job
  - [ ] [How to use SSH plugin](https://github.com/microsoft/pai/tree/master/src/kube-runtime/src/plugins/ssh)
  - [ ] [How to enable SSH for a docker image](https://github.com/microsoft/pai/blob/master/docs/job_docker_env.md#enable-ssh-for-your-image)
  - [ ] [How to use TensorBoard plugin](https://github.com/microsoft/pai/blob/master/docs/user/tensorboard.md)
- [ ] How to Manage Data (***need help***)
  - [ ] Team-wise storage - NFS
  - [ ] Team-wise storage - Samba
  - [ ] Team-wise storage - AzureFile
  - [ ] Team-wise storage - AzureBlob
- [ ] Advanced Jobs
  - [ ] Retry policy and [job exit spec](https://github.com/microsoft/pai/blob/master/src/k8s-job-exit-spec/config/user-manual.md) (***need help***)
  - [ ] Job example with multiple task roles
     - [ ] [tensorflow_cifar10](https://github.com/microsoft/pai/blob/master/marketplace-v2/tensorflow-cifar10.yaml)
     - [ ] [horovod_pytorch](https://github.com/microsoft/pai/blob/master/marketplace-v2/horovod-pytorch-synthetic-benchmark.yaml)
  - [ ] [Port reservation , environmental variables, parameters, and secrets](https://github.com/microsoft/pai/blob/master/docs/user/job_submission.md#use-parameters-and-secrets)
  - [ ] RDMA job (***need help***)
  - [ ] Infinityband job (***need help***)
  - [ ] [Reference link to job protocol](https://github.com/microsoft/openpai-protocol/blob/master/schemas/v2/schema.yaml)
- [ ] Use Marketplace
  - [ ] Brief usage introduction (***need help***)
- [ ] Use VSCode extension
  - [ ] [Brief usage introduction](https://github.com/microsoft/openpaivscode)
  - [ ] [Reference link to VSCode extension](https://github.com/microsoft/openpaivscode)
- [ ] Use SDK
  - [ ] [Brief Usage introduction](https://github.com/microsoft/openpaisdk)
  - [ ] [Reference link to SDK](https://github.com/microsoft/openpaisdk)

### Content of cluster administrator manual

- [ ] How to Install OpenPAI
  - [ ] [From scratch - quick start script](https://github.com/microsoft/pai/blob/master/contrib/kubespray/doc/quick-start.md)
  - [ ] [From previous deployment - update guide](https://github.com/microsoft/pai/blob/master/docs/upgrade/upgrade_to_v0.18.0.md)
- [ ] Basic Management
  - [ ] Brief introduction to webportal functions
    - [ ] Hardware utilization page
    - [ ] Services page
    - [ ] User management
    - [ ] Abnormal jobs
    - [ ] [How to access k8s dashboard](https://github.com/microsoft/pai/blob/master/docs/k8s-dashboard/access-k8s-dashboard.md)
  - [ ] [How to manage user group and virtual cluster](https://github.com/wutb15/pai/blob/b80c8dffd8e028e0013366eb0cc7d622ae0c9917/docs/user_group_management.md) (***need help***)
- [ ] How to Add / Remove Node (***need help***)
- [ ] PAI service management and paictl
  - [ ] Brief introduction to PAI services
  - [ ] [Use paictl to customize config, and restart PAI services](https://github.com/microsoft/pai/blob/master/docs/paictl/paictl-manual.md)
  - [ ] [How to setup webportal plugin](https://github.com/microsoft/pai/blob/d659ffd97dedfc23ce1b06eebc2911b9c6592ac7/docs/webportal/PLUGINS.md)
- [ ] How to Setup Team-wise Storage  (***need help***)
  - [ ] NFS server (storage manager)
  - [ ] Samba 
  - [ ] AzureFile
  - [ ] Azureblob
- [ ] Advanced Installation Guide (***need help***)
  - [ ] CPU (GPU) Node
  - [ ] Multi-type GPUs and VCs
- [ ] [How to Setup AAD](https://github.com/microsoft/pai/blob/master/docs/aad-e2e/aad-e2e.md)
- [ ]  How to Uninstall OpenPAI
- [ ]  Troubleshooting (***need help***)
  - [ ] What if GPU is not detected
  - [ ] What if some node is lost  
  - [ ] What if some PAI service is not working
- [ ] Upgrade Guide (needed in future)

### Content of PAI developer Manual
 
- [ ] [OpenPAI Basic Design and Architecture](https://github.com/microsoft/pai/blob/141301d4ad637cbe0fff41e40e60bf7bf2797900/docs/system_architecture.md)
  - [ ] How OpenPAI uses Nvidia-docker
  - [ ] How OpenPAI uses Kubernetes
- [ ] Lifetime Management of OpenPAI Jobs (***need help***)
- [ ] How OpenPAI Manages Storage
- [ ] Network Management of Mixed Cloud Environment (***need help***)
- [ ] How to Develop a Runtime Plugin (***need help***)
- [ ] [How to Develop a Webportal Plugin](https://github.com/microsoft/pai/blob/d659ffd97dedfc23ce1b06eebc2911b9c6592ac7/docs/webportal/PLUGINS.md#provide-plugin-configurations)
- [ ] How to Configure Framework Controller (***need help***)
- [ ] How to Customize Job Protocol  (***need help***)
- [ ] How to Develop a Scheduler (***need help***)
